app/components/ConfirmTrade.tsx

```diff
@@ -0,0 +1,327 @@
+import MarketABI from "@/abi/market.json";
+import { CONDITIONAL_TOKEN_CONTRACT_ADDRESS } from "@/model/conditionalTokens";
+import ConditionalTokensABI from "@/abi/conditionalTokens.json";
+import { addFraction, removeFraction } from "@/utils/price";
+import { useConfig } from "wagmi";
+import { Address, erc20Abi, formatEther, parseEther } from "viem";
+import { useReadCalcSellAmount } from "@/model/market";
+import { WXADI } from "@/constants";
+
+const ROUNDING_PRECISON = 0.00000000001;
+
+interface ConfirmTradeProps {
+  swapState: SwapState;
+  marketId: Address;
+  onApprove: Function;
+  onSwap: Function;
+  outcomeIndex: number;
+  swapDirection: SwapDirection;
+  tokenAmountIn: string;
+  tokenAmountOut?: bigint;
+}
+
+export const ConfirmTrade = ({
+  swapState,
+  marketId,
+  onApprove,
+  onSwap,
+  outcomeIndex,
+  swapDirection,
+  tokenAmountIn,
+  tokenAmountOut,
+}: ConfirmTradeProps) => {
+  const config = useConfig();
+  const { isModalOpen, openModal, closeModal } = useModalContext();
+  const [txHash, setTxHash] = useState("");
+  const [isTxLoading, setIsTxLoading] = useState(false);
+  const { data: sellAmount } = useReadCalcSellAmount(
+    marketId,
+    formatEther(tokenAmountOut || BigInt(0)),
+    outcomeIndex
+  );
+
+  const closeBetModal = () => {
+    closeModal(ModalId.CONFIRM_SWAP);
+  };
+
+  const closeTxModal = () => {
+    closeModal(ModalId.WAITING_TRANSACTION);
+  };
+  const openTxModal = () => {
+    openModal(ModalId.WAITING_TRANSACTION);
+  };
+
+  const amountWei = parseEther(tokenAmountIn);
+  const twoDecimalsTokenAmountIn = tokenAmountIn
+    ? parseFloat(tokenAmountIn).toFixed(2)
+    : "";
+  const twoDecimalsTokenAmountOut = tokenAmountOut
+    ? parseFloat(formatEther(tokenAmountOut)).toFixed(2)
+    : "";
+
+  const approveToken = async () => {
+    setIsTxLoading(true);
+
+    try {
+      const txHash = await writeContract(config, {
+        abi: erc20Abi,
+        address: WXADI.address,
+        functionName: "approve",
+        args: [marketId, amountWei],
+      });
+      setTxHash(txHash);
+      openTxModal();
+
+      await waitForTransactionReceipt(config, {
+        hash: txHash,
+      });
+      onApprove();
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsTxLoading(false);
+    }
+  };
+
+  const approveNFT = async () => {
+    setIsTxLoading(true);
+    try {
+      const txHash = await writeContract(config, {
+        abi: ConditionalTokensABI,
+        address: CONDITIONAL_TOKEN_CONTRACT_ADDRESS,
+        functionName: "setApprovalForAll",
+        args: [marketId, true],
+      });
+      setTxHash(txHash);
+      openTxModal();
+
+      await waitForTransactionReceipt(config, {
+        hash: txHash,
+      });
+      onApprove();
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setIsTxLoading(false);
+    }
+  };
+
+  const submitBet = async () => {
```

Comment:
```suggestion
  const submitBuyBet = async () => {
```

this rename indicates the method "buy" it makes it consisten with `submitSellBet`